android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java

```diff
@@ -6,69 +6,26 @@
-import android.widget.TextView;
 
 import org.chromium.base.Callback;
-import org.chromium.chrome.R;
-import org.chromium.ui.modaldialog.DialogDismissalCause;
 import org.chromium.ui.modaldialog.ModalDialogManager;
-import org.chromium.ui.modaldialog.ModalDialogManager.ModalDialogType;
-import org.chromium.ui.modaldialog.ModalDialogProperties;
-import org.chromium.ui.modaldialog.SimpleModalDialogController;
-import org.chromium.ui.modelutil.PropertyModel;
 
+/**
+ *
+ * This is override chromium android 13 default dialog
+ * */
 public class BraveNotificationPermissionRationaleDialogController
         extends NotificationPermissionRationaleDialogController {
-    private final ModalDialogManager mModalDialogManager;
-    private final Context mContext;
-
     public BraveNotificationPermissionRationaleDialogController(
             Context context, ModalDialogManager modalDialogManager) {
         super(context, modalDialogManager);
-        mModalDialogManager = modalDialogManager;
-        mContext = context;
     }
 
+    /**
+     * This is empty override method, to block default android 13 chromium dialog
+     * */
     @Override
-    public void showRationaleUi(Callback<Integer> rationaleCallback) {
-        LayoutInflater inflater = LayoutInflater.from(mContext);
-        Resources resources = mContext.getResources();
-
-        View dialogView = inflater.inflate(R.layout.brave_notification_permission_rationale_dialog,
-                /* root= */ null);
-        TextView titleView = dialogView.findViewById(R.id.notification_permission_rationale_title);
-        TextView descriptionView =
-                dialogView.findViewById(R.id.notification_permission_rationale_message);
-
-        PropertyModel.Builder dialogModelBuilder =
-                new PropertyModel.Builder(ModalDialogProperties.ALL_KEYS)
-                        .with(ModalDialogProperties.CONTROLLER,
-                                new SimpleModalDialogController(mModalDialogManager,
-                                        wrapDialogDismissalCallback(rationaleCallback)))
-                        .with(ModalDialogProperties.CUSTOM_VIEW, dialogView);
-
-        PropertyModel dialogModel = dialogModelBuilder.build();
-
-        // continue button
-        Button btnContinue = dialogView.findViewById(R.id.notification_continue_button);
-        btnContinue.setOnClickListener((v) -> {
-            mModalDialogManager.dismissDialog(
-                    dialogModel, DialogDismissalCause.POSITIVE_BUTTON_CLICKED);
-        });
-
-        // now Button
-        Button btnNotNow = dialogView.findViewById(R.id.notification_not_now_button);
-        btnNotNow.setOnClickListener((v) -> {
-            mModalDialogManager.dismissDialog(
-                    dialogModel, DialogDismissalCause.NEGATIVE_BUTTON_CLICKED);
-        });
-
-        mModalDialogManager.showDialog(dialogModel, ModalDialogType.APP);
-    }
+    public void showRationaleUi(Callback<Integer> rationaleCallback) {}
```

Comment:
do you want to remove this code here?

Reply:
in 3rd launch 
1. can remove request notification permission
2. request notification permission instead of showing chromium dialog request through OS dialog

Let me know which one you prefer